internal/gitinterface/hash.go

```diff
@@ -0,0 +1,54 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package gitinterface
+
+import (
+	"encoding/hex"
+	"errors"
+)
+
+const (
+	zeroSHA1HashString   = "0000000000000000000000000000000000000000"
+	zeroSHA256HashString = "0000000000000000000000000000000000000000000000000000000000000000"
+)
+
+var (
+	ErrInvalidHashEncoding = errors.New("hash string is not hex encoded")
+	ErrInvalidHashLength   = errors.New("hash string is wrong length")
+)
+
+// Hash represents a Git object hash. It is a lightweight wrapper around the
+// standard hex encoded representation of a SHA-1 or SHA-256 hash used by Git.
+type Hash struct {
```

Comment:
Is this used anywhere yet? We may want to hold off on this change for the time being.

Reply:
It is used in the next PR onwards for all git IDs, just to sanity check that anything we're passing to a git command (say cat-file or whatever) is a valid SHA.

Reply:
I'm using this PR to set up the foundations for the object specific APIs introduced in the next PRs in this series